pandas/core/dtypes/cast.py

```diff
@@ -1314,6 +1315,30 @@ def find_result_type(left_dtype: DtypeObj, right: Any) -> DtypeObj:
         #  which will make us upcast too far.
         if lib.is_float(right) and right.is_integer() and left_dtype.kind != "f":
             right = int(right)
+        # After NEP 50, numpy won't inspect Python scalars
+        # TODO: do we need to recreate numpy's inspection logic for floats too
+        # (this breaks some tests)
+        if isinstance(right, int) and not isinstance(right, np.integer):
+            # This gives an unsigned type by default
+            # (if our number is positive)
+
+            # If our left dtype is unsigned, we might not want this since
+            # this might give us 1 dtype too big, though.
+            # W should check if the corresponding int dtype (e.g. int64 for uint64)
```

Comment:
Doesn't taking the negative of a potentially unsigned number break this?

```python
>>> val = 2 ** 64 - 1
>>> np.min_scalar_type(val)
dtype('uint64')
>>> np.min_scalar_type(-val)
dtype('O')
```

Reply:
So, if I exclude object dtype, this should work then?

cc @seberg for another pair of eyes

Reply:
Do we have test(s) that hit this directly? If not I think worth setting up to be explicit